specification/cognitiveservices/HealthInsights/healthinsights.common/primitives.tsp

```diff
@@ -4,14 +4,19 @@ namespace AzureHealthInsights;
 
 #suppress "@azure-tools/typespec-azure-core/long-running-polling-operation-required" "This is a template"
 @doc("Long running RPC operation template")
+@post
 op LongRunningRpcOperation<
-  TParams,
-  TResponse
+  TParams extends TypeSpec.Reflection.Model,
+  TResponse extends TypeSpec.Reflection.Model
 > is Azure.Core.Foundations.Operation<
   TParams & RepeatabilityRequestHeaders,
-  (Foundations.AcceptedResponse<Foundations.LongRunningStatusLocation &
-    Foundations.RetryAfterHeader> &
-    RepeatabilityResponseHeaders) | TResponse,
+  Foundations.AcceptedResponse<{
+    @pollingLocation
+    @doc("The location for monitoring the operation state.")
+    @TypeSpec.Http.header("Operation-Location")
+    operationLocation: TypeSpec.Rest.ResourceLocation<TResponse>;
+  } & Foundations.RetryAfterHeader &
+    RepeatabilityResponseHeaders> | TResponse,
   RepeatabilityRequestHeaders & RepeatabilityResponseHeaders,
   Azure.Core.Foundations.ErrorResponse
```

Comment:
There were behavior change in TCGC `getLroMetadata` on previous template https://github.com/Azure/azure-rest-api-specs/blob/2bfcaad/specification/cognitiveservices/HealthInsights/healthinsights.common/primitives.tsp

Before 0.32 (maybe even earlier), the `logicResult` is `TResponse`.
Since 0.32, the `logicResult` changed to `OperationStatus<never, Error>` (which is the model referred in `Foundations.LongRunningStatusLocation`).

To unblock emitter/SDK that depends on the TCGC and the healthinsights, I had to push the fix to template, ahead of https://github.com/Azure/typespec-azure/issues/3108

However, I had to add `@lroStatus` to the `status` property in `Response`. It not, the `logicResult` be `undefined`.

And then, I had to add `Canceled` to the status enum. If not, there is lint error.